extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/reactive/ReactiveMongoCollection.java

```diff
@@ -0,0 +1,1410 @@
+     *
+     * @return the class
+     */
+    Class<T> getDocumentClass();
+
+    /**
+     * Gets an estimate of the count of documents in a collection using collection metadata.
+     *
+     * @return n {@link Uni} completed with the estimated number of documents
+     */
+    Uni<Long> estimatedDocumentCount();
+
+    /**
+     * Gets an estimate of the count of documents in a collection using collection metadata.
+     *
+     * @param options the options describing the count
+     * @return n {@link Uni} completed with the estimated number of documents
+     */
+    Uni<Long> estimatedDocumentCount(EstimatedDocumentCountOptions options);
+
+    /**
+     * Counts the number of documents in the collection.
+     *
+     * @return n {@link Uni} completed with the number of documents
+     */
+    Uni<Long> countDocuments();
+
+    /**
+     * Counts the number of documents in the collection according to the given options.
+     *
+     * @param filter the query filter
+     * @return n {@link Uni} completed with the number of documents
+     */
+    Uni<Long> countDocuments(Bson filter);
+
+    /**
+     * Counts the number of documents in the collection according to the given options.
+     *
+     * @param filter the query filter
+     * @param options the options describing the count
+     * @return n {@link Uni} completed with the number of documents
+     */
+    Uni<Long> countDocuments(Bson filter, CountOptions options);
+
+    /**
+     * Counts the number of documents in the collection according to the given options.
+     *
+     * @param clientSession the client session with which to associate this operation
+     * @return n {@link Uni} completed with the number of documents
+     */
+    Uni<Long> countDocuments(ClientSession clientSession);
+
+    /**
+     * Counts the number of documents in the collection according to the given options.
+     *
+     * @param clientSession the client session with which to associate this operation
+     * @param filter the query filter
+     * @return n {@link Uni} completed with the number of documents
+     */
+    Uni<Long> countDocuments(ClientSession clientSession, Bson filter);
+
+    /**
+     * Counts the number of documents in the collection according to the given options.
+     *
+     * @param clientSession the client session with which to associate this operation
+     * @param filter the query filter
+     * @param options the options describing the count
+     * @return n {@link Uni} completed with the number of documents
+     */
+    Uni<Long> countDocuments(ClientSession clientSession, Bson filter, CountOptions options);
+
+    /**
+     * Gets the distinct values of the specified field name.
+     *
+     * @param fieldName the field name*
+     * @param clazz the default class to cast any distinct items into.
+     * @param <D> the target type of the iterable.
+     * @return a multi emitting the sequence of distinct values
```

Comment:
You add a link to `Uni` in the Javadoc and not to `Multi` ?

Reply:
Fixed.